graphql_compiler/compiler/emit_sql.py

```diff
@@ -1,7 +1,115 @@
+    # changes due to collapsing into a CTE.
+    'query_path_to_output_fields',
+    # 'compiler_metadata': CompilerMetadata, SQLAlchemy metadata about Table objects, and
+    # further backend specific configuration.
+    'compiler_metadata',
+))
 
 
 def emit_code_from_ir(sql_query_tree, compiler_metadata):
-    """Return a SQLAlchemy Query from a passed tree representation of an SQL query."""
-    raise NotImplementedError(u'SQL query emitting is not yet supported.')
+    """Return a SQLAlchemy Query from a passed SqlQueryTree.
+
+    Args:
+        sql_query_tree: SqlQueryTree, tree representation of the query to emit.
+        compiler_metadata: CompilerMetadata, SQLAlchemy specific metadata.
+
+    Returns: SQLAlchemy Query
+
+    """
+    context = CompilationContext(
+        query_path_to_selectable=dict(),
+        query_path_to_location_info=sql_query_tree.query_path_to_location_info,
+        query_path_to_output_fields=sql_query_tree.query_path_to_output_fields,
+        compiler_metadata=compiler_metadata,
+    )
+
+    return _query_tree_to_query(sql_query_tree.root, context)
+
+
+def _query_tree_to_query(node, context):
+    """Convert this node into its corresponding SQL representation.
+
+    Args:
+        node: The node to convert to SQL.
+        context: CompilationContext, compilation specific metadata
+
+    Returns:
+        Query, the compiled SQL query
+    """
+    _create_table_and_update_context(node, context)
+    return _create_query(node, context=context)
+
+
+def _create_table_and_update_context(node, context):
+    """Create an aliased table for a SqlNode.
+
+    Updates the relevant Selectable global context.
+
+    Args:
+        node: SqlNode, the current node.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Table, the newly aliased SQLAlchemy table.
+    """
+    schema_type_name = context_helpers.get_schema_type_name(node, context)
+    table = context.compiler_metadata.get_table(schema_type_name).alias()
+    context.query_path_to_selectable[node.query_path] = table
+    return table
+
+
+def _create_query(node, context):
+    """Create a query from a SqlNode.
+
+    Args:
+        node: SqlNode, the current node.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Selectable, selectable of the generated query.
+    """
+    output_columns = _get_output_columns(node, context)
+    selectable = context_helpers.get_node_selectable(node, context)
+    query = select(output_columns).select_from(selectable)
+    return query
+
+
+def _get_output_columns(node, context):
+    """Get the output columns required by the query.
+
+    Args:
+        node: SqlNode, the current node.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        List[column], list of SqlAlchemy columns to output for this query.
```

Comment:
is `column` the name of the type? Should it be `Column` instead?

Reply:
Yeah the base class is Column, but imported and used as column.